src/py/flwr/client/process/process.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower background ClientApp."""
+
+from logging import DEBUG, ERROR, INFO
+from typing import Tuple
+
+import grpc
+
+from flwr.client.client_app import ClientApp
+from flwr.common import Context, Message
+from flwr.common.grpc import create_channel
+from flwr.common.logger import log
+from flwr.common.serde import (
+    context_from_proto,
+    context_to_proto,
+    message_from_proto,
+    message_to_proto,
+    run_from_proto,
+)
+from flwr.common.typing import Run
+
+# pylint: disable=E0611
+from flwr.proto.clientappio_pb2 import (
+    PullClientAppInputsRequest,
+    PullClientAppInputsResponse,
+    PushClientAppOutputsRequest,
+    PushClientAppOutputsResponse,
+)
+from flwr.proto.clientappio_pb2_grpc import ClientAppIoStub
+
+from .utils import _get_load_client_app_fn
+
+
+def _run_background_client(  # pylint: disable=R0914
+    address: str,
+    token: int,
+) -> None:
+    """Run background Flower ClientApp process.
+
+    Parameters
+    ----------
+    address : str
+        Address of SuperNode
+    token : int
+        Unique SuperNode token for ClientApp-SuperNode authentication
+    """
+
+    def on_channel_state_change(channel_connectivity: str) -> None:
+        """Log channel connectivity."""
+        log(DEBUG, channel_connectivity)
+
+    channel = create_channel(
+        server_address=address,
+        insecure=True,
+    )
+    channel.subscribe(on_channel_state_change)
+
+    try:
+        stub = ClientAppIoStub(channel)
+
+        # Pull Message, Context, and Run from SuperNode
+        message, context, run = pull_message(stub=stub, token=token)
```

Comment:
What happens if there's no message?

Reply:
I assume you're referring to the case where there is no message on the SuperNode to be pulled. There is a if-statement in `client/app.py` [here](https://github.com/adap/flower/blob/ead952b7a2a8218d2c70801cbc7977fe088dbf4c/src/py/flwr/client/app.py#L369-L371) that continues the while loop if message is `None`. 

The other possibility is that (i) the `message_to_proto` serde failed, which should be avoidable since we have a test for that, or (ii) there is a packet loss. I'm not sure about the latter but I think an error is raised there by gRPC.

Reply:
ok to leave it as-is for now. we'll have to address it when we continuously pull the next inputs.